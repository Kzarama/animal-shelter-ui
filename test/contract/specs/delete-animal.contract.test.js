import { AnimalController } from "../../../controllers";
import { provider } from "../config/initPact";

describe("Animal Service", () => {
  beforeAll(async () => {
    await provider.setup();
  });

  describe("When a request to delete an animal is made", () => {
    beforeAll(async () => {
      await provider.addInteraction({
        uponReceiving: "a request to delete an animal",
        state: "delete animal",
        withRequest: {
          method: "DELETE",
          path: "/animals/Manchas",
        },
        willRespondWith: {
          status: 204,
        },
      });
    });

    test("should return the correct data", async () => {
      const response = await AnimalController.delete("Manchas");
      expect(response.data).toMatchSnapshot();
      await provider.verify();
    });
  });
  afterAll(() => provider.finalize());
});
