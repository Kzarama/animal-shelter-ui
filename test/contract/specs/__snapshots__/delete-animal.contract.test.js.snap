// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Animal Service When a request to delete an animal is made should return the correct data 1`] = `""`;
